network/manager.go

```diff
@@ -698,3 +686,28 @@ func (nm *networkManager) GetEndpointID(containerID, ifName string) string {
 	}
 	return containerID + "-" + ifName
 }
+
+func cnsEndpointInfotoCNIEpInfo(endpointInfo restserver.EndpointInfo, endpointID string) *EndpointInfo {
+	epInfo := &EndpointInfo{
+		Id:                 endpointID,
+		IfIndex:            EndpointIfIndex, // Azure CNI supports only one interface
+		ContainerID:        endpointID,
+		PODName:            endpointInfo.PodName,
+		PODNameSpace:       endpointInfo.PodNamespace,
+		NetworkContainerID: endpointID,
+	}
+
+	for ifName, ipInfo := range endpointInfo.IfnameToIPMap {
+		if ifName != InfraInterfaceName {
+			// TODO: filling out the SecondaryNICs from the state for Swift 2.0
+			continue
+		}
+		// filling out the InfraNIC from the state
+		epInfo.IPAddresses = ipInfo.IPv4
```

Comment:
Instead of comparing with string for infraNIC, should we compare with NICtype ?

Reply:
I did not do that since the migrated states do not have the NICType there and we can only rely on Ifname for now.